resource/labeling-app/components/text-sample/textArea.js

```diff
@@ -281,6 +281,10 @@ const TextArea = {
             }
         });
 
+        document.getElementsByClassName('sample-container')[0].addEventListener('click', (event) => {
```

Comment:
The fact that your component knows who his parent is is not very nice. Ideally it should be isolated so that if we change 'sample-container' to anything else we don't need to change child components. To achieve that you can assign a reference to your text area:

```
<TextArea v-if="type === 'text'"
                            ref="textarea"
                            :text="item.data.raw[config.text_column]"
...
```

and you can call any method of TextArea from the parent like:
`$refs.textarea.deselectAll()`

So I'd create an extra background element to capture outside clicks:
```
<div class="sample-container-background" @click="$refs.textarea.deselectAll()"></div>
```
it should be a first child of `<div class="sample-container">` before other components.

the style could be this for example:

```
.sample-container-background {
    position: absolute;
    width: 100%;
    height: 100%;
}
```

In this case TextArea just exposes it's API and it parent's job to call it

Reply:
That is indeed a way better way to handle it, thanks, I'll fix that.

Reply:
Fixed in 417e43b. I did it a little bit differently, I did not add any div, I used the one already existing, but I created a function that handles the click. I had to make an exception if the click is in the textarea as I couldn't even select a word otherwise (The deselectAll was applying all the times"